clients/destination.go

```diff
@@ -264,6 +264,10 @@ func (c *DestinationClient) Write(ctx context.Context, source string, syncTime t
 			Source:    source,
 			Timestamp: timestamppb.New(syncTime),
 		}); err != nil {
+			if err == io.EOF {
```

Comment:
```suggestion
			if errors.Is(err, io.EOF) {
```
`errors.Is` is much more reliable as the errors are wrapped more and more often.